src/transport/raw/BLE.h

```diff
@@ -91,6 +100,14 @@ class DLL_EXPORT BLEBase : public Base, public Ble::BleLayerDelegate
 
     CHIP_ERROR SetEndPoint(Ble::BLEEndPoint * endPoint) override;
 
+    /**
+     * Change BLE transport to this
+     *
+     * This is relevant when a device is both a commissioner and a commissionee
+     * and so BLE transport needs to be able to toggle between these
+     */
+    void OverrideTransport() { mBleLayer->mBleTransport = this; }
```

Comment:
Without the doxycomment, the code of `foo->OverrideTransport` is unclear. Could we name it something that does not require doccomment reads on 'what transport? override with what?'.

I suggest `SetBleLayerTransportToSelf` or provide a `& GetBleLayerTransport()` method and then code can read `foo->GetBleLayerTransport() = foo` or some other way (like `auto layer = foo->GetBleLayer(); layer->mBleTransport = foo`).

The lifetime of this override is unclear to me because it is circular:
  - mBleLayer is a member of BLEBase
  - mBLELayer will have a member pointing to BLEBase

Reply:
Reading the delta now, I see that this circular dependency is not new. 

However reading more: where is this `OverrideTransport` method called?

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/15525

added comments explained rationale and behavior options